libs/cosmos-sdk/baseapp/abci.go

```diff
@@ -403,6 +406,29 @@ func handleSimulate(app *BaseApp, path []string, height int64, txBytes []byte, o
 	if err != nil {
 		return sdkerrors.QueryResult(sdkerrors.Wrap(err, "failed to decode tx"))
 	}
+	msgs := tx.GetMsgs()
+	if enableFastQuery() && len(msgs) > 0 && msgs[0].Route() == "wasm" {
```

Comment:
if multi msgs，simulate gas will not enough

Reply:
OK